### PR TITLE
LPS-88204 add tests to check if the empty option is shown properly

### PR DESCRIPTION
**Pull request by marcosapmf**
Related Issue: https://issues.liferay.com/browse/LPS-88204

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
Just started reviewing :)<br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js [hunk 1]
```diff
@@ -488,6 +500,7 @@ const Select = ({
 						handleSelect={handleSelect}
 						multiple={multiple}
 						options={options}
+						showEmptyOption={showEmptyOption}
 					/>
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js:504]: @marcosapmf porque só passou o parametro pro `< DropdownListWithSearch` e não pro `<DropdownList`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js [hunk 2]
```diff
@@ -262,13 +269,17 @@ const DropdownListWithSearch = ({
...
-		const result = options.filter(
+		let result = options.filter(
 			(option) =>
 				option.value &&
 				option.label.toLowerCase().includes(query.toLowerCase())
 		);
 
-		setFilteredOptions([emptyOption, ...result]);
+		if (showEmptyOption) {
+			result = [emptyOption, ...result];
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js:279]: @marcosapmf seria bom colocar o `emptyOption ` dentro do `if` como tu fez no outro caso.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleEditor/RuleEditor.soy [hunk 3]
```diff
@@ -673,6 +675,7 @@
 							{/param}
 							{param ref: 'firstOperand' + $index /}
 							{param spritemap: $spritemap /}
+							{param showEmptyOption: false /}
 							{param value: [$firstOperandValueSelected] /}
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleEditor/RuleEditor.soy:679]: @marcosapmf colocar antes do `spritemap `

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js (hunk 1 above)
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Select/Select.es.js:504]: @carolmariaabb, `DropdownListWithSearch` adiciona o seu próprio empty option, por isso foi necessário colocar o parametro

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward
